app/src/main/java/org/oppia/app/profile/AddProfileActivity.kt

```diff
@@ -40,4 +40,8 @@ class AddProfileActivity : InjectableAppCompatActivity() {
     super.onActivityResult(requestCode, resultCode, data)
     addProfileFragmentPresenter.handleOnActivityResult(requestCode, resultCode, data)
   }
+
+  override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
```

Comment:
Actually, it stops the system from saving the state and restoring it. This allows me to handle the state manually on the configuration change.  
I had to do it this way because there is a bug related to the view binding. If I type a name only in the `input_name` and make a configuration change, the first 3 letters are input in both `input_pin` and `input_confirm_pin`. I think that happens because of the custom view which has the same id for the `EditText` in each view.  
I'm not sure how `ViewBinding` handles this, but I think it is related to it.